app/models/mhv_account.rb

```diff
@@ -69,7 +82,11 @@ def terms_and_conditions_accepted?
   end
 
   def preexisting_account?
-    user&.mhv_correlation_id.present?
+    user&.mhv_correlation_id.present? && !previously_registered?
+  end
+
+  def accessible?
+    upgraded? || existing?
```

Comment:
if the account was `created` but not upgraded, it would be in a `failed_upgrade` state, but should that maybe still have access to everything that created should have access to? something to think about.

Reply:
yes, this is true, but not if the check_eligibility triggers "existing." If you recall, that particular transition will return once the very first transition triggers true. So I'm thinking it needs to exclude failed upgrade and other states, since those should continue on and get "upgraded."

Reply:
right now you have "existing" as the first check in check_eligibility.

Reply:
I think this has been resolved on Slack; `preexisting_account?` checks for `registered_at`, so a an account that's already been registered through Vets.gov and has an MHV id can't fall into the `existing` state. In any case, I've added a set of specs for this set of transitions from a state with MHV id so we can be sure.